### PR TITLE
[WIP] *: support read statistic upload to pd

### PR DESCRIPTION
**Pull request by choleraehyq**
@zhangjinpeng1987 PTAL

Signed-off-by: Cholerae Hu <huyingqian@pingcap.com>

### CODE REVIEW

File: src/coprocessor/metrics.rs [hunk 1]
```diff
@@ -85,4 +88,10 @@ lazy_static! {
             "Total number of rocksdb query of get or scan count",
             &["type"]
         ).unwrap();
+
+    // region id => read bytes of this region
+    pub static ref COPR_READ_BYTES: Mutex<HashMap<u64, u64>> =  Mutex::new(HashMap::new());
```
Comment (siddontang) [inline src/coprocessor/metrics.rs]: as I said in the discussion doc, don't use static var, bad style.

File: src/storage/mvcc/reader.rs [hunk 2]
```diff
@@ -96,6 +96,7 @@ impl<'a> MvccReader<'a> {
         };
 
         self.statistics.data.processed += 1;
+        self.statistics.data.read_bytes += res.len() as u64;
```
Comment (zhangjinpeng1987) [inline src/storage/mvcc/reader.rs]: Add key.len() to statistics too.

File: src/storage/mvcc/reader.rs [hunk 3]
```diff
@@ -109,13 +110,19 @@ impl<'a> MvccReader<'a> {
 
         let res = if let Some(ref mut cursor) = self.lock_cursor {
             match try!(cursor.get(key, &mut self.statistics.lock)) {
-                Some(v) => Some(try!(Lock::parse(v))),
+                Some(v) => {
+                    self.statistics.lock.read_bytes += v.len() as u64;
```
Comment (zhangjinpeng1987) [inline src/storage/mvcc/reader.rs]: No need to statistics read bytes/keys for lock cf.

File: src/coprocessor/metrics.rs (hunk 1 above)
Comment (siddontang) [inline src/coprocessor/metrics.rs]: said again, remove it.

**Pull request closed without merging**